sympy/vector/coordsysrect.py

```diff
@@ -438,6 +438,12 @@ def orient_new(self, name, orienters, location=None,
                 final_matrix = orienters.rotation_matrix(self)
             else:
                 final_matrix = orienters.rotation_matrix()
+            # TODO: trigsimp is needed here so that the matrix becomes
+            # canonical (scalar_map also calls trigsimp; without this, you can
+            # end up with the same CoordinateSystem that compares differently
+            # due to a differently formatted matrix). However, this is
+            # probably not so good for performance.
+            final_matrix = trigsimp(final_matrix)
```

Comment:
This is probably really bad for performance. I'd prefer not to have it in there, but we can merge this and figure that out later. I don't think the new vector package is used for any performance critical tasks yet, e.g. we don't use it for the mechanics package.